projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/use-uploader.js

```diff
@@ -22,6 +22,107 @@ export const getJWT = function ( key ) {
 const jwtsForKeys = {};
 
 export const resumableUploader = ( { onError, onProgress, onSuccess } ) => {
+	const GUID_HEADER = 'x-videopress-upload-guid';
+	const MEDIA_ID_HEADER = 'x-videopress-upload-media-id';
+	const SRC_URL_HEADER = 'x-videopress-upload-src-url';
+
+	const extractUploadKeyForUrl = urlString => {
+		const url = new URL( urlString );
+		const path = url.pathname;
+		const parts = path.split( '/' );
+		return parts[ parts.length - 1 ];
+	};
+
+	const getJwtForKey = ( maybeUploadkey, method ) => {
+		return new Promise( ( resolve, reject ) => {
+			if ( jwtsForKeys[ maybeUploadkey ] ) {
+				resolve( jwtsForKeys[ maybeUploadkey ] );
+			} else if ( 'HEAD' === method ) {
+				getJWT( maybeUploadkey )
+					.then( responseData => {
+						jwtsForKeys[ maybeUploadkey ] = responseData.token;
+						resolve( jwtsForKeys[ maybeUploadkey ] );
+					} )
+					.catch( reject );
+			} else {
+				reject( 'No jwt present' );
+			}
+		} );
+	};
+
+	const extractHeadersFromXhr = res => {
+		const guid = res.getHeader( GUID_HEADER );
+		const mediaId = res.getHeader( MEDIA_ID_HEADER );
+		const src = res.getHeader( SRC_URL_HEADER );
+		return { src, mediaId, guid };
+	};
+
+	const extractHeadersFromFetch = res => {
+		const guid = res.headers.get( GUID_HEADER );
+		const mediaId = res.headers.get( MEDIA_ID_HEADER );
+		const src = res.headers.get( SRC_URL_HEADER );
+		return { src, mediaId, guid };
+	};
+
+	const triggerSuccessWhenGuidHeaderPresent = headers => {
+		const { src, mediaId, guid } = headers;
+		if ( guid && mediaId && src ) {
+			onSuccess && onSuccess( { mediaId: Number( mediaId ), guid, src } );
+			return true;
+		}
+
+		return false;
+	};
+
+	const methodNeedsUploadKey = method =>
+		[ 'OPTIONS', 'GET', 'HEAD', 'DELETE', 'PUT', 'PATCH' ].indexOf( method ) >= 0;
+
+	const startPollingForGuid = url => {
+		console.log( url );
+		// recursiveUrlRequest
+		const numRetries = 10;
+		const waitMs = 1000;
+		let timerId = null;
+		let retries = 0;
+		const recursiveUrlRequest = function recursiveUrlRequest() {
+			if ( retries > numRetries ) {
+				timerId && clearTimeout( timerId );
+				onError && onError();
```

Comment:
```suggestion
				onError && onError( 'Guid polling exceeded number of retries' );
```

I got an exception when testing and this line was hit. I think `onError` requires something to be passed to it.